sycl/unittests/scheduler/ErrcUnboundPlaceholder.cpp

```diff
@@ -0,0 +1,86 @@
+//==-------- ErrcUnbouncPlaceholder.cpp ------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
```

Comment:
@sergey-semenov  - can I get a consultation?

Getting back to this PR, I've added this unit test, and it works on Linux.  It has a very minimal mock.  But on Windows I'm getting a CL_INVALID_BINARY error.  At first I thought perhaps just calling `setupDefaultMockAPIs()` would fix it, but no such luck.

FWIW, I don't quite understand why redefining piDeviceGetInfo to be successful does the trick on Linux, but it does and the kernel is enqueued, the exception is thrown, the errc tested, etc. Awesome. 

But not on Windows. and I'm not sure what the best course is.  I can do tricks to get the unit test to succeed (like return PI_DEVICE_TYPE_ACC for any PI_DEVICE_INFO_TYPE request), but I think that ends up just with the test not actually running - though it's hard to say on Win.  

Also, FWIW, there seem to be environment dependencies that make working on these tests in isolation difficult (or make achieving repeatability difficult). But that's not the immediate concern.